autoload.py

```diff
@@ -21,12 +21,14 @@
 SCRIPT_TYPE_GDB = "gdb"
 SCRIPT_TYPE_LLDB = "lldb"
 
-# gdb uses the script name to avoid running the same script twice. This set
-# allows us to do the same.
-loaded_scripts = set()
-# Sometimes lldb will give us the same module in multiple events. We use this
-# set to deduplicate them.
-modules_processed = set()
+# gdb uses the script name to avoid running the same script twice. This
+# dictionary is used to avoid loading the same script for the same debugger
+# instance twice.
+loaded_scripts = {}
+# Sometimes lldb will give us the same module in multiple events. This
+# dictionary is used to avoid loading the same module for the same debugger
+# instance twice.
+modules_processed = {}
 modules_loaded_callbacks = []
```

Comment:
I think we should also make `modules_loaded_callbacks` have a list for each debugger, and then pass the debugger object to the callback.

Reply:
Does the function `register_modules_loaded_callback` ever get called? I don't see where this is called nor seeing any logging from this function if I added one.

Reply:
I updated this register_modules_loaded_callback to be a dictionary from debugger id to list. Is this what you want?

Reply:
Yes, thank you